services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -872,12 +947,55 @@ private Runnable getRunnableForDeferredVersionSwap() {
             break;
           }
 
+          ThreadPoolExecutor clusterExecutorService = getOrCreateExecutorForCluster(cluster);
+          ThreadPoolStats clusterThreadPoolStats = clusterToThreadPoolStatsMap.get(cluster);
+
+          // Filter out stores that aren't doing a target region push w/ deferred swap
+          List<Store> eligibleStoresToProcess = new ArrayList<>();
+          for (Store parentStore: parentStores) {
+            if (!isTargetRegionPushWithDeferredSwapEnabled(parentStore)) {
+              continue;
+            }
+            eligibleStoresToProcess.add(parentStore);
+          }
+
           boolean sequentialRollForward = !StringUtils.isEmpty(rolloutOrderStr);
-          if (sequentialRollForward) {
-            performSequentialRollForward(cluster, parentStores, childControllerClientMap, rolloutOrder);
-          } else {
-            performParallelRollForward(cluster, parentStores, childControllerClientMap);
+          for (Store parentStore: eligibleStoresToProcess) {
+            Version targetVersion = parentStore.getVersion(parentStore.getLargestUsedVersionNumber());
+            if (targetVersion == null) {
```

Comment:
You already added these checks in `isTargetRegionPushWithDeferredSwapEnabled`, may be skip them here?